crates/iceberg-catalog/src/service/endpoint_statistics.rs

```diff
@@ -0,0 +1,271 @@
+        request_metadata: RequestMetadata,
+        response_status: StatusCode,
+        path_params: HashMap<String, String>,
+        query_params: HashMap<String, String>,
+    },
+    Shutdown,
+}
+
+#[derive(Debug, Default)]
+pub struct ProjectStatistics {
+    stats: HashMap<EndpointIdentifier, AtomicI64>,
+}
+
+impl ProjectStatistics {
+    #[must_use]
+    pub fn into_consumable(self) -> HashMap<EndpointIdentifier, i64> {
+        self.stats
+            .into_iter()
+            .map(|(k, v)| (k, v.load(std::sync::atomic::Ordering::Relaxed)))
+            .collect()
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct EndpointIdentifier {
+    pub uri: Endpoints,
+    pub status_code: StatusCode,
+    pub warehouse: Option<WarehouseIdent>,
+    // probably only relevant for config calls
+    pub warehouse_name: Option<String>,
+}
+
+#[derive(Debug)]
+pub struct EndpointStatisticsTracker {
+    rcv: tokio::sync::mpsc::Receiver<EndpointStatisticsMessage>,
+    endpoint_statistics: HashMap<ProjectIdent, ProjectStatistics>,
+    statistic_sinks: Vec<Arc<dyn EndpointStatisticsSink>>,
+    flush_interval: Duration,
+}
+
+impl EndpointStatisticsTracker {
+    #[must_use]
+    pub fn new(
+        rcv: tokio::sync::mpsc::Receiver<EndpointStatisticsMessage>,
+        stat_sinks: Vec<Arc<dyn EndpointStatisticsSink>>,
+        flush_interval: Duration,
+    ) -> Self {
+        Self {
+            rcv,
+            endpoint_statistics: HashMap::new(),
+            statistic_sinks: stat_sinks,
+            flush_interval,
+        }
+    }
+
+    pub async fn run(mut self) {
+        let mut last_update = tokio::time::Instant::now();
+        loop {
+            if last_update.elapsed() > self.flush_interval {
+                tracing::debug!(
+                    "Flushing stats after: {}ms",
+                    last_update.elapsed().as_millis()
+                );
+                self.flush_storage().await;
+                last_update = tokio::time::Instant::now();
+            }
+
+            let Ok(msg) = tokio::time::timeout(self.flush_interval, self.rcv.recv()).await else {
+                tracing::debug!("No message received, continuing.");
+                continue;
+            };
+
+            let Some(msg) = msg else {
+                tracing::info!("Channel closed, shutting down.");
+                self.close().await;
+                break;
+            };
+
+            tracing::debug!("Received message: {:?}", msg);
+
+            match msg {
+                EndpointStatisticsMessage::EndpointCalled {
+                    request_metadata,
+                    response_status,
+                    path_params,
+                    query_params,
+                } => {
+                    let warehouse = Self::maybe_get_warehouse_ident(&path_params);
+
+                    let Some(matched_path) = request_metadata.matched_path() else {
+                        tracing::error!("No path matched.");
```

Comment:
Is this worth an error log? I assume that happens quite often with typos etc.
I am not sure that we want to print this as error.
I think I would use something below info.